AutofacContrib.NSubstitute.Tests/ExampleFixture.cs

```diff
@@ -193,5 +193,17 @@ public void Example_test_with_substitute_for_concrete_resolved_from_autofac()
 
             Assert.That(result, Is.EqualTo(val2 * val3 * 2));
         }
+
+        [Test]
+        public void ConfigureContainer()
+        {
+            var container = Substitute.For<IContainer>();
+
+            using var mock = AutoSubstitute.Configure()
+                .ConfigureContainerBuilder(b => container)

```

Comment:
I'm still a bit uncertain about this. If you put your own container in then it looses the other things within AutoSubstitute - it means that there is little point in using it.

If the builder is a `Func<ContainerBuilder, ContainerBuilder>` that makes more sense to me. Can you achieve your use case with that?

Reply:
What would it lose? For instance, in my use case, I use the MultitenantContainer, which wraps an already built container. So I would do something like:

```
using var mock = AutoSubstitute.Configure()
   .ConfigureContainerBuilder(b => new MultitenantContainer(b.Build(), ...))
   .Build()
```

It still uses the container configured by AutoSubstitute, but just wraps it in another layer. This way it gets passed to any of the post-configure steps (ie the ProvidedValue setting)

Reply:
I've dealt with the merge conflicts and switched it so that it is a parameter passed to the `Build(...)`. The scenario I'm trying to solve really needs to be able to hook into the actual creation of the resulting `IContainer` before the provided values are set.